examples/llamacpp_processor.py

```diff
@@ -0,0 +1,68 @@
+from enum import Enum
+
+import numpy as np
+from llama_cpp import Llama, LogitsProcessorList, StoppingCriteria, StoppingCriteriaList
+from numpy.typing import NDArray
+from pydantic import BaseModel, constr
+
+from outlines.generate.processors.llamacpp import JSONLogitsProcessor
+
+
+class Weapon(str, Enum):
+    sword = "sword"
+    axe = "axe"
+    mace = "mace"
+    spear = "spear"
+    bow = "bow"
+    crossbow = "crossbow"
+
+
+class Armor(str, Enum):
+    leather = "leather"
+    chainmail = "chainmail"
+    plate = "plate"
+
+
+class Character(BaseModel):
+    name: constr(max_length=10)
+    age: int
+    armor: Armor
+    weapon: Weapon
+    strength: int
+
+
+# TODO: why do we need this?
+class EosCriteria(StoppingCriteria):
+    def __init__(self, eos_token_id):
+        self.eos_token_id = eos_token_id
+
+    def __call__(self, input_ids: NDArray[np.intc], logits: NDArray[np.single]):
+        if self.eos_token_id in input_ids[1:]:
+            return True
+
+
+if __name__ == "__main__":
+    llama = Llama("./phi-2.Q4_K_M.gguf")
+
+    prompt = b"Instruct: You are a leading role play gamer. You have seen thousands of different characters and their attributes.\nPlease return a JSON object with common attributes of an RPG character. Give me a character description\nOutput:"
+
+    logits_processor = JSONLogitsProcessor(Character, llama)
+    stopping_criteria_list = StoppingCriteriaList([EosCriteria(llama.token_eos())])
+
+    json_str = ""
+    tokens = llama.tokenize(prompt)
+    for token in llama.generate(
```

Comment:
Does `llama.generate` return an iterator or are you just iterating over the results?

Reply:
It's a generator and we should probably use `llama.create_completion` here...